lib/new_relic/agent/instrumentation/redis/chain.rb

```diff
@@ -9,16 +9,28 @@ def self.instrument!
         ::Redis::Client.class_eval do
```

Comment:
For prepend you're doing 2 different classes based on redis version, ::RedisClient  ::Redis::Client. How come that isn't necessary here? I would think that we'd be prepending/chaining on the same class always, so I'm curious about how this is working.

Reply:
Fantastic question! After taking a closer look, it seems like the Redis 5 instrumentation does not need to use the `redis-client` library (`::RedisClient`) as the entrypoint for chain/prepend after all! I thought it needed to be the source because I couldn't find a definition for `connect` in the `::Redis::Client` class, however, after looking at the caller locations for both chain and prepend using only `::Redis::Client`, it all still works! I'll go back to using ::Redis::Client for both since the change seems unnecessary.

Here's the relevant locations for chain if you're interested: 

```
 "~/dev/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/redis/instrumentation.rb:16:in `connect_with_tracing'",
 "~/dev/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/redis/chain.rb:39:in `connect'",
 "~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/redis-client-0.11.1/lib/redis_client.rb:647:in `raw_connection'",
 "~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/redis-client-0.11.1/lib/redis_client.rb:614:in `ensure_connected'",
 "~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/redis-client-0.11.1/lib/redis_client.rb:213:in `call_v'",
 "~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/redis-5.0.5/lib/redis/client.rb:73:in `call_v'",
 "~/dev/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/redis/chain.rb:16:in `block in call_v'",
 "~/dev/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/redis/instrumentation.rb:50:in `block in call_v_with_tracing'",
 "~/dev/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/redis/instrumentation.rb:65:in `block in with_tracing'",
 "~/dev/newrelic-ruby-agent/lib/new_relic/agent/tracer.rb:355:in `capture_segment_error'",
 "~/dev/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/redis/instrumentation.rb:65:in `with_tracing'",
 "~/dev/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/redis/instrumentation.rb:50:in `call_v_with_tracing'",
 "~/dev/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/redis/chain.rb:16:in `call_v'",
 "~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/redis-5.0.5/lib/redis.rb:167:in `block in send_command'",
 "~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/redis-5.0.5/lib/redis.rb:166:in `synchronize'",
 "~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/redis-5.0.5/lib/redis.rb:166:in `send_command'",
 "~/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/redis-5.0.5/lib/redis/commands/strings.rb:191:in `get'",
 ```

Reply:
Resolved in: 57320913dbd719ce65656e6db97dfe5f80d7a4c4